pydantic/import_helper.py

```diff
@@ -0,0 +1,33 @@
+class V2MigrationRemoved(Exception):
+    ...
+
+
+class V2MigrationMoved(Exception):
+    ...
+
+
+class V2MigrationRenamed(Exception):
+    ...
+
+
+class V2MigrationSuperseded(Exception):
+    ...
+
+
+V2_MIGRATION_MAPPING = {
+    # Case: Removed
+    "pydantic.Required": V2MigrationRemoved("pydantic.Required has been removed in favour of ..."),
+    # Case: Moved path
+    # Case: Renamed object
+    # Case: Superseded
+}
+
+import importlib
```

Comment:
I think clearer to just implement `__getattr__` in `__init__.py`.

Reply:
The only was I could get `__getattr__` to work was on each file that was being accessed, so `from pydantic import Required` would be fine in `__init__.py` but `from pydantic.generics import replace_types` would require the `__getattr__` in `pydantic/generics.py` too

Reply:
I think so this should go into a function and be called in `__init__.py`, not run on module import.

I'm still not sure about this route, is it supported by all supported versions and pypy?

Reply:
I'll go towards how it is now and start building up the scenarios, standards for messages etc and then can revisit this part if its not ideal. However, with the way it is at the moment it is isolated which has its advantages